volumes/repository.go

```diff
@@ -170,6 +170,7 @@ func (r *Repository) Delete(path string) error {
 	}
 
 	if volume.IsBindMount {
+		r.remove(volume)
```

Comment:
Nitpick, but I would find it more readable if the `IsBindMount` condition was surrounding the `r.driver.Remove` call, and rely on the final (unconditional) `r.remove(volume)`, as in:

```
if !volume.isBindMount {
    if err := r.driver.Remove(volume.ID); err != nil {
        if !os.IsNotExist(err) {
            return err
        }
    }

    r.remove(volume)
    return nil
}
```